### PR TITLE
Unset IP addresses on deprovisioning 

### PR DESCRIPTION
**Pull request by bjormel**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by bjormel** [comment]
@mpolden I hope that this is sufficient

**Review by mpolden — APPROVED** [review]
LGTM! What is the reason we need to do this?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java [hunk 1]
```diff
@@ -629,8 +629,10 @@ public Node markNodeAvailableForNewAllocation(String hostname, Agent agent, Stri
                 List<Node> removed = new ArrayList<>(children);
                 if (zone.cloud().value().equals("aws"))
                     db.removeNodes(List.of(node));
-                else
+                else {
+                    node = node.with(IP.Config.EMPTY);
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java:633]: Consider extending one of the deprovision tests in `NodeRepositoryTest` to test this behaviour.

**Comment by mpolden** [comment]
.

**Pull request merged**